From 7c8f82ac449bc8ec85681546fc7f9a9abf1284e8 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Mon, 10 Jun 2019 18:40:09 +0300
Subject: [PATCH 285/386] net: mvpp2: fix long poll flow control for jumbo
 frames

New port->pkt_size wasn't set yet, so code checked old pkt_size
value.

Change-Id: I0e554f588a5ba343e78912200bb4819286ff75a8
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/10688
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 7b029507fbe6..89fb0277760c 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -1194,7 +1194,7 @@ static int mvpp2_bm_update_mtu(struct net_device *dev, int mtu)
 					mvpp2_bm_pool_update_fc(port,
 								pools_pcpu[cpu],
 								false);
-			} else if (port->pkt_size > MVPP2_BM_LONG_PKT_SIZE)
+			} else if (pkt_size > MVPP2_BM_LONG_PKT_SIZE)
 				mvpp2_bm_pool_update_fc(port,
 							port->pool_short,
 							false);
@@ -1229,7 +1229,7 @@ static int mvpp2_bm_update_mtu(struct net_device *dev, int mtu)
 					mvpp2_bm_pool_update_fc(port,
 								pools_pcpu[cpu],
 								false);
-			} else if (port->pkt_size > MVPP2_BM_LONG_PKT_SIZE)
+			} else if (pkt_size > MVPP2_BM_LONG_PKT_SIZE)
 				mvpp2_bm_pool_update_fc(port, port->pool_long,
 							true);
 			else
-- 
2.17.1

